### PR TITLE
Support for Terraform init, and using backend configs.

### PR DESCRIPTION
**Pull request by jaustinpage**
# Features:

This adds support for terraform init, as well as use of backend configs. 
Fixes https://github.com/beelit94/python-terraform/issues/11

## Detailed Notes:

This extends the locations that Terraform checks to include .terraform/terraform.tfstate, as well as provides handling of "backend_config" dictionaries for init commands.

I also added tests to handle finding of the tfstate file when none is provided and the default ./terraform.tfstate is present and finding the tfstate file when none is provided and the backend tfstate is in use (./.terraform/terraform.tfstate).

Let me know if there is anything I can do to make this better! Thanks!

### CODE REVIEW

**Comment by jaustinpage** [comment]
It appears that your develop branch is a little bit behind your master branch. I went ahead and rebased my changes on master, so that there are no conflicts. If you would prefer, I can base my changes on develop instead.

**Comment by beelit94** [comment]
@jaustinpage Thank you for your awesome PR, We'll look into it ASAP

File: python_terraform/__init__.py [hunk 1]
```diff
@@ -134,6 +134,35 @@ def plan(self, dir_or_plan=None, detailed_exitcode=IsFlagged, **kwargs):
         args = self._generate_default_args(dir_or_plan)
         return self.cmd('plan', *args, **options)
 
+    def init(self, dir_or_plan=None, backend_config=None,
+            reconfigure=IsFlagged, force_copy=IsNotFlagged, backend=True,
+            **kwargs):
+        """
```
[the remark this answers is not in the document]
Comment (jaustinpage) [reply python_terraform/__init__.py]: Sure! Feel free to make the change, thanks!
